### PR TITLE
Global error explanations improvements

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

**Comment by Manishearth** [comment]
@bors r+

I don't think we should be rewrapping the text in future PRs though.

**Comment by GuillaumeGomez** [comment]
@Manishearth: I did it only when I found something to change in the file.

**Comment by Manishearth** [comment]
Most of this PR is rewrapped paragraphs with no other change.

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -104,28 +104,86 @@ E0004: r##"
 This error indicates that the compiler cannot guarantee a matching pattern for
 one or more possible inputs to a match expression. Guaranteed matches are
 required in order to assign values to match expressions, or alternatively,
-determine the flow of execution.
+determine the flow of execution. Erroneous code example:
+
+```compile_fail
```
Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: @Manishearth: I added code examples in this file.

File: src/librustc_borrowck/diagnostics.rs [hunk 2]
```diff
@@ -73,7 +74,14 @@ fn main() {
 ```
 
 To fix this, ensure that any declared variables are initialized before being
-used.
+used. Example:
+
+```
```
Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: @Manishearth: I added code example here as well.

File: src/librustc_privacy/diagnostics.rs [hunk 3]
```diff
@@ -165,7 +165,7 @@ let f = Bar::Foo(0); // error: cannot invoke tuple struct constructor with
 ```
 
 To solve this issue, please ensure that all of the fields of the tuple struct
-are public. Alternatively, provide a new() method to the tuple struct to
+are public. Alternatively, provide a `new()` method to the tuple struct to
```
Comment (GuillaumeGomez) [inline src/librustc_privacy/diagnostics.rs]: @Manishearth: I added backquotes in here.

File: src/librustc/diagnostics.rs (hunk 1 above)
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: ?

Scroll down.

File: src/librustc_resolve/diagnostics.rs [hunk 4]
```diff
@@ -941,7 +943,7 @@ use something::self; // ok!
 "##,
 
 E0431: r##"
-`self` import was made. Erroneous code example:
+An invalid `self` import was made. Erroneous code example:
```
Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: @Manishearth: I improved the sentence.

**Comment by GuillaumeGomez** [comment]
@Manishearth: So in conclusion, the rewrapping has only be done on diagnostics files which does have something else to change.

**Comment by Manishearth** [comment]
@GuillaumeGomez yes, my point was, don't do that. There's no clear benefit of either rewrapping choice, and this just causes unnecessary churn to review and fill the history.

**Pull request merged**